projects/canopy/src/lib/utils/dynamic-style.service.ts

```diff
@@ -0,0 +1,97 @@
+import { Injectable } from '@angular/core';
+
+import { BreakpointValues } from '../shared/breakpoints.interface';
+
+interface Rule {
+  selector: string;
+  declaration: string;
+}
+interface RuleRepsonive {
+  selector: string;
+  declaration: string;
+  breakpoint: BreakpointValues;
+}
+
+@Injectable({
+  providedIn: 'root',
+})
+/**
+ * Dynamically add CSS rules to a <style> tag in the <head>
+ */
+export class DynamicStyleService {
+  styleTag = null;
+  selectors = [];
+  mediaQueries = {};
+
+  private addStyleTag(): void {
+    this.styleTag = document.createElement('style');
+    this.styleTag.type = 'text/css';
+    document.getElementsByTagName('head')[0].appendChild(this.styleTag);
+  }
+
+  private createMediaQuery(breakpoint: BreakpointValues): string {
+    if (this.mediaQueries[breakpoint]) {
+      return;
+    }
+    this.mediaQueries[breakpoint] = [];
+    return `@media(min-width:${breakpoint}){}`;
+  }
+
+  addRules(rules: Array<Rule>): void {
+    if (!this.styleTag) {
+      this.addStyleTag();
+    }
+
+    // cache current styles to avoid multiple edits to the innerHTML
+    let styleTagCache = this.styleTag.innerHTML.slice();
+
+    rules.map((r) => {
+      if (this.selectors.includes(r.selector)) {
+        return;
+      }
+
+      this.selectors.push(r.selector);
+
+      const ruleString = `.${r.selector}{${r.declaration}!important}`;
+      styleTagCache += ruleString;
```

Comment:
```suggestion
      styleTagCache += `.${r.selector}{${r.declaration}!important}`;
```
it's just styling preference, but you could leave the assignment.

Reply:
Nice, yeah it's a little cleaner this way, done ✅